### ISSUE TITLE
error in set focus

### ISSUE THREAD

**Issue by julientype** [mtxr/vscode-sqltools#146]
**Describe the bug**
when i click view table  it renders the data window.....
the selected entry point is highlighted the table name ...ok
the error return creates a second highlighted the data connection entry..... bad error
it forces the selected entry to jump up the the data connection....
I have small screen so it mess up my tree on a large screen the effects would be un notice

next is add 2 new icons for the tree one yellow key for primary and one orange  for foreign key 

i added this new feature because your playing with the tree view and you can fix both at the same time....

**To Reproduce**
[sqlite]
**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Windows, Linux, Mac]
 - SQLTools Version [e.g. v0.17.x]

**Additional context**
styles and features go hand and hand .....
https://www.iconfinder.com/
basically you setup your styles  the table icon and field type icons   
the icon-finder will let you build the icon  you need as temp icons ......
a text icon a blob icon an integer icon...

**Comment by mtxr** [comment]
Is it the latest version @julientype ?

**Comment by julientype** [comment]
yes the view was corrected as you did....  the return focus created 2  focus points where only 1  should be at all times base on the operation at hand.... styles to new features go hand and hand.... this way your not back tracking to correct features later..... so you have a style error and the selected action return entry
like if you create something what return focus entry should highlighted in the tree  we cant have multi entries can we....   

stay tune on on multi dimension form layout for sql .... you can  create forms where your trying to create arrays to build sql formats in javascripts memory or let dom do it for you..... xml is an loaded array with speed...

**Comment by julientype** [comment]
ohh yes their is away to query the css being used in its frame to get its tree active object.....   again your not building arrays to cycle an active request .... your working from the load dom object... and let it do the work for you....   you do get dom access in vs-code addins but its limited  granted by vscode styles themes.....   your working off its style object.....

**Comment by julientype** [comment]
Get a white board as you read draw the ideas on the board on how we are going to move information around..... your a data store connector and the cat window is your client......  
the client will never connect to the database .... your the connector store.... where you add alter values from to the database.... like add delete create.....  where the complex code is not mix in with data connection store......
with client creating sql do you need the store to create it ...no.....  so we create a server client apps this way and delegate operation to one or the other..... 
do you need a grid layout yes..... do i no..... i need data sachems injections from the store to pay with

**Comment by mtxr** [comment]
Wait. We are talking about a lot of stuff.

What's the problem? Is because the 3 is being expanded on connect?

**Comment by julientype** [comment]
in a client server operation we need maintain a connection  so its important the data store tree to retain 1 element selected ..... you have a cat window open and moving to different   area in the store....  
a) do you close the cat window
b) let the user submit sql   from a client window.  ..no
your command operation happens by the user.... im not sure  how vscode lets you do this in its implemented styles .... but styles are styles....  
 you do need to master it single or multi selector in css and multi is harder where a request operation is sent to a corrected selected item or items ...lol.....for its focus set or get element at hand....
css   
a:focus {
     color:rgb(85, 171, 228);
}

.divToolTree,
a:hover {
    color:rosybrown;
}

.divToolTree,
a:active {
    color: olive;
}

**Comment by julientype** [comment]
here is the code.... where a long array gets mess up and you need the user back to its inserted operation in the tree....
elem[iclause].scrollIntoView({});

**Comment by mtxr** [comment]
Ok, I got it now.

But currently there's nothing I can do with vscode styles. Their api programmatically create an item defining icons and some other props, but it's JS only. CSS is not allowed right now. There is an issue on VSCode repo about this, I don't remember the name. I'll relate it later here, but for now, my hands are tied. Some extensions change VScode styles, but it's not persistente. The user need to re-do this whenever it updates

**Comment by mtxr** [comment]
I'll close this issue since it's not a SQLTools bug itself. But I'll open or thumbs up the related issue on VSCode repository.

Thanks @julientype !

**Comment by mtxr** [comment]
@julientype can you create a PR for this feature?
